lib/tag.js

```diff
@@ -60,6 +60,27 @@ var tag = function tag(tagName, attrsMap, content, contentIsEscaped) {
     return '<' + safeTagName + attrsHTML + (isSelfClosing(safeTagName) ? ' />' : '>' + safeContent + '</' + safeTagName + '>');
 };
 
+// expose
 tag.attrs = attrs;
 
+// --------------------
+
+var dataRegExp = /^data-[a-z]+([-][a-z]+)*$/;
+var ariaRegExp = /^aria-[a-z]+$/;
+var legalAttrs = ['autocomplete', 'autocorrect', 'autofocus', 'autosuggest', 'checked', 'dirname', 'disabled', 'tabindex', 'list', 'max', 'maxlength', 'min', 'novalidate', 'pattern', 'placeholder', 'readonly', 'required', 'size', 'step'];
+var ignoreAttrs = ['id', 'name', 'class', 'classes', 'type', 'value', 'multiple'];
+var getUserAttrs = function (opt) {
+    return reduce(opt, function (attributes, option, k) {
+        if ((ignoreAttrs.indexOf(k) === -1 && legalAttrs.indexOf(k) > -1) || dataRegExp.test(k) || ariaRegExp.test(k)) {
+            attributes[k] = option;
+        }
+        return attributes;
+    }, {});
+};
+
+// expose
+tag.getUserAttrs = getUserAttrs;
```

Comment:
If you want to expose a function, then it's worth looking at the code to understand its use cases - especially when that enables you to write tests.